File: php/features.php
<?php
declare(strict_types=1);

// Microlink SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MicrolinkFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MicrolinkBaseFeature();
            case "test":
                return new MicrolinkTestFeature();
            default:
                return new MicrolinkBaseFeature();
        }
    }
}

File: py/features.py
# Microlink SDK feature factory

from feature.base_feature import MicrolinkBaseFeature
from feature.test_feature import MicrolinkTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MicrolinkBaseFeature(),
        "test": lambda: MicrolinkTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
